tests/components/asuswrt/test_sensor.py

```diff
@@ -27,16 +32,56 @@
         CONF_PROTOCOL: "ssh",
         CONF_USERNAME: "fake_user",
         CONF_PASSWORD: "fake_pass",
-        CONF_SENSORS: "upload",
+        CONF_SENSORS: [
+            "devices",
+            "download_speed",
+            "download",
+            "upload_speed",
+            "upload",
+        ],
     }
 }
 
+Device = namedtuple("Device", ["mac", "ip", "name"])
 
-async def test_default_sensor_setup(hass):
+MOCK_DEVICES = {
+    "a1:b1:c1:d1:e1:f1": Device("a1:b1:c1:d1:e1:f1", "192.168.1.2", "Test"),
+    "a2:b2:c2:d2:e2:f2": Device("a2:b2:c2:d2:e2:f2", "192.168.1.3", "TestTwo"),
+    "a3:b3:c3:d3:e3:f3": Device("a3:b3:c3:d3:e3:f3", "192.168.1.4", "TestThree"),
+}
+MOCK_BYTES_TOTAL = [60000000000, 50000000000]
+MOCK_CURRENT_TRANSFER_RATES = [20000000, 10000000]
+
+
+async def test_sensors(hass: HomeAssistant):
     """Test creating an AsusWRT sensor."""
     with patch("homeassistant.components.asuswrt.AsusWrt") as AsusWrt:
         AsusWrt().connection.async_connect = mock_coro_func()
+        AsusWrt().async_get_connected_devices = mock_coro_func(MOCK_DEVICES)
+        AsusWrt().async_get_bytes_total = mock_coro_func(MOCK_BYTES_TOTAL)
+        AsusWrt().async_get_current_transfer_rates = mock_coro_func(
+            MOCK_CURRENT_TRANSFER_RATES
+        )
 
-        result = await async_setup_component(hass, DOMAIN, VALID_CONFIG_ROUTER_SSH)
-        assert result
+        assert await async_setup_component(hass, DOMAIN, VALID_CONFIG_ROUTER_SSH)
+        await hass.async_block_till_done()
         assert hass.data[DATA_ASUSWRT] is not None
+
+        now = datetime(2020, 1, 1, 1, tzinfo=dt_util.UTC)
+        with patch(("homeassistant.helpers.event.dt_util.utcnow"), return_value=now):
```

Comment:
I think we should patch time before we set up the component, to make sure we control time from the beginning.
```suggestion
        now = datetime(2020, 1, 1, 1, tzinfo=dt_util.UTC)
        with patch(("homeassistant.helpers.event.dt_util.utcnow"), return_value=now):
            assert await async_setup_component(hass, DOMAIN, VALID_CONFIG_ROUTER_SSH)
            await hass.async_block_till_done()
```
We should also avoid assert things about hass.data, as that is a detail of the integration that may change in the future.